src/main/java/org/testng/internal/TestInvoker.java

```diff
@@ -102,7 +105,11 @@ public ITestResultNotifier getNotifier() {
               testMethod, System.currentTimeMillis(), new Throwable(okToProceed));
       m_notifier.addSkippedTest(testMethod, result);
       InvokedMethod invokedMethod = new InvokedMethod(System.currentTimeMillis(), result);
-      invokeListenersForSkippedTestResult(result, invokedMethod);
+      try {
+        invokeListenersForSkippedTestResult(result, invokedMethod);
+      } catch (SkipException e) {
```

Comment:
The catch is not placed in a good place. Check the listeners and you'll see that `afterInvocation` won't be called as expected.

Reply:
And something is weird: why is the test already skipped? `// Not okToProceed. Test is being skipped`
Additional question: how can it be skipped if the catch of the exception does nothing?